src/betterproto/__init__.py

```diff
@@ -601,6 +603,43 @@ def _get_cls_by_field(
         return field_cls
 
 
+HybridT = TypeVar("HybridT")
+
+
+class hybridmethod(Generic[HybridT]):
+    """Hybrid method decorator.
+
+    Can be used to have both classmethod and instance method with the same name and
+    execute the right one depending on context.
+
+    Source: https://stackoverflow.com/a/28238047
+    """
+
+    def __init__(
+        self,
+        fclass: Type[HybridT],
+        finstance: Optional[HybridT] = None,
+        doc: Optional[str] = None,
+    ):
+        self.fclass = fclass
+        self.finstance = finstance
+        self.__doc__ = doc or fclass.__doc__
+        # support use on abstract base classes
+        self.__isabstractmethod__ = bool(getattr(fclass, "__isabstractmethod__", False))
+
+    def classmethod(self, fclass: Type[HybridT]):
+        return type(self)(fclass, self.finstance, None)
+
+    def instancemethod(self, finstance: HybridT):
+        return type(self)(self.fclass, finstance, self.__doc__)
+
+    def __get__(self, instance: Optional[T], cls: Type[T]):
+        if instance is None or self.finstance is None:
+            # either bound to the class, or no instance method available
+            return self.fclass.__get__(cls, None)
+        return self.finstance.__get__(instance, cls)
+
+
```

Comment:
This file is starting to get very bloated, would you be able to pull this into a utils file or similar